### PR TITLE
octopus: rgw: fix bucket purge incomplete multipart uploads

### PR DESCRIPTION
**Pull request by ivancich**
backport tracker: https://tracker.ceph.com/issues/53213

parent tracker (pacific): https://tracker.ceph.com/issues/52662

---

The marker was not working correctly as segments of the bucket index
were listed to shut down any incomplete multipart uploads. This fixes
the marker, so it's maintained properly across iterations.

Signed-off-by: J. Eric Ivancich <ivancich@redhat.com>
Fixes: https://tracker.ceph.com/issues/53213

Note: this is currently a place-holder. When pacific version is merged, we'll cherry-pick from there to document the chain.

### CODE REVIEW

**Comment by ivancich** [comment]
@smithfarm,

This is the octopus version of the fix. I'll document the cherry-pick and conflicts once the pacific version is merged. So for now it's a placeholder.

Don't know if we should update the tracker to indicate it's pacific rather than octopus? Your guidance would be appreciated.

**Review by adamemerson — APPROVED** [review]

**Review by smithfarm — CHANGES_REQUESTED** [review]
do not open "placeholder" PRs. They might get merged before they get updated to their final state.

After #43862 gets merged, this needs to be redone using `git cherry-pick -x`

**Comment by smithfarm** [comment]
> Don't know if we should update the tracker to indicate it's pacific rather than octopus?

Doesn't the tracker have an "affected versions" field?

You already have "Backport: octopus" in the tracker. That's pretty clear.

**Comment by ivancich** [comment]
> do not open "placeholder" PRs. They might get merged before they get updated to their final state.
> 
> After #43862 gets merged, this needs to be redone using `git cherry-pick -x`

I believe I've followed this process. Please let me know if there are any issues. Thanks for your help!

**Comment by smithfarm** [comment]
This is a nit, but could you move the conflicts section down below the "cherry picked from" line?

It seems useful to have a convention where everything above the "cherry picked from" line is from the original commit and everything below it is specific to the cherry-pick.

**Comment by smithfarm** [comment]
> I believe I've followed this process.

If you wait until the master PR (pacific PR in this case) gets merged, you don't have to re-do anything, and no reviewers need to get alarmed that a bad cherry-pick might make it into a mature stable branch.

**Comment by ivancich** [comment]
> > I believe I've followed this process.
> 
> If you wait until the master PR (pacific PR in this case) gets merged, you don't have to re-do anything, and no reviewers need to get alarmed that a bad cherry-pick might make it into a mature stable branch.

I understand, and maybe I was unclear. I was just letting you know that I did things to follow the process you'd outlined. And I've fixed the ordering in the commit. Thanks!

**Comment by yuriw** [comment]
jenkins test api

**Pull request merged**
